ui/v2.5/src/components/Shared/StashBoxIDSearchModal.tsx

```diff
@@ -0,0 +1,445 @@
+            </div>
+          )}
+        </div>
+      </Row>
+      <Row>
+        <Col>
+          <TruncatedText text={scene.details ?? ""} lineCount={3} />
+        </Col>
+      </Row>
+      <SearchResultTags tags={scene.tags} />
+    </div>
+  );
+};
+
+export const SceneSearchResult: React.FC<ISceneResultProps> = ({ scene }) => {
+  return (
+    <div className="mt-3 search-item">
+      <SceneSearchResultDetails scene={scene} />
+    </div>
+  );
+};
+
+// Main Modal Component
+export const StashBoxIDSearchModal: React.FC<IProps> = ({
+  entityType,
+  stashBoxes,
+  onHide,
+  onSelectItem,
+}) => {
+  const intl = useIntl();
+  const Toast = useToast();
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const [selectedStashBox, setSelectedStashBox] = useState<GQL.StashBox | null>(
+    null
+  );
+  const [query, setQuery] = useState<string>("");
+  const [results, setResults] = useState<
+    | GQL.ScrapedPerformerDataFragment[]
+    | GQL.ScrapedStudioDataFragment[]
+    | GQL.ScrapedSceneDataFragment[]
+  >([]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    if (stashBoxes.length > 0) {
+      setSelectedStashBox(stashBoxes[0]);
+    }
+  }, [stashBoxes]);
+
+  useEffect(() => inputRef.current?.focus(), []);
+
+  const doSearch = useCallback(async () => {
+    if (!selectedStashBox || !query) {
+      return;
+    }
+
+    setLoading(true);
+    setResults([]);
+
+    try {
+      let queryData;
+
+      switch (entityType) {
+        case "performer":
+          queryData = await stashBoxPerformerQuery(
+            query,
+            selectedStashBox.endpoint
+          );
+          setResults(queryData.data?.scrapeSinglePerformer ?? []);
+          break;
+        case "studio":
+          queryData = await stashBoxStudioQuery(
+            query,
+            selectedStashBox.endpoint
+          );
+          setResults(queryData.data?.scrapeSingleStudio ?? []);
+          break;
+        case "scene":
+          queryData = await stashBoxSceneQuery(
+            query,
+            selectedStashBox.endpoint
+          );
+          setResults(queryData.data?.scrapeSingleScene ?? []);
+          break;
+      }
+    } catch (error) {
+      Toast.error(error);
+    } finally {
+      setLoading(false);
+    }
+  }, [query, selectedStashBox, entityType, Toast]);
+
+  function handleItemClick(
+    item:
+      | GQL.ScrapedPerformerDataFragment
+      | GQL.ScrapedStudioDataFragment
+      | GQL.ScrapedSceneDataFragment
+  ) {
+    if (selectedStashBox && item.remote_site_id) {
+      onSelectItem(item.remote_site_id, selectedStashBox.endpoint);
+      onHide();
+    }
+  }
+
+  function renderResults() {
+    if (results.length === 0) {
+      return null;
+    }
+
+    return (
+      <div className={CLASSNAME_LIST_CONTAINER}>
+        <div className="mt-1 mb-2">
+          <FormattedMessage
+            id={`dialogs.${entityType}s_found`}
```

Comment:
Having an `s` suffix here isn't going to work for all locales. Look at the `countables` entries in `en-GB.json` for guidance.